sw/host/cryptotest/testvectors/parsers/nist_cavp_aes_gcm_parser.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Parser for converting NIST AES-GCM testvectors to JSON.
+
+"""
+
+import argparse
+import sys
+import json
+import jsonschema
+
+from cryptotest_util import parse_rsp, str_to_byte_array
+
+
+def parse_testcases(args) -> None:
+    raw_testcases = parse_rsp(args.src)
+    testcases = list()
+    for test_case_id, test_vec in enumerate(raw_testcases):
+        testcase = {
+            "test_case_id": test_case_id,
+            "vendor": "nist",
+            "mode": "gcm",
+            "operation": args.operation.lower(),
+            "key_len": args.key_len,
+            "key": str_to_byte_array(test_vec["Key"]),
+            "aad": str_to_byte_array(test_vec["AAD"]),
+            "iv": str_to_byte_array(test_vec["IV"]),
+            "ciphertext": str_to_byte_array(test_vec["CT"]),
+            "plaintext": str_to_byte_array(test_vec["PT"]) if "PT" in test_vec else [],
+            "result": True if "PT" in test_vec else False,
```

Comment:
Why is the expected result "pass" for all decryption tests and "failure" for all encryption tests?

Reply:
Looks like a mistake from my side while working from another work setup. That's unfortunate that somehow it slipped through.
Currently, there are no FAIL cases in the GCM encryption test vectors. However, for decryption test vectors, if a test vector lacks a Plaintext (PT) field, the result of that particular test vector will be marked as FAIL.

Thanks a lot for pointing this out! I am sending a PR with the fix.

Reply:
#21881 21881